sync/class.jetpack-sync-module-posts.php

```diff
@@ -316,76 +286,84 @@ function filter_post_content_and_add_links( $post_object ) {
 	}
 
 	public function save_published( $new_status, $old_status, $post ) {
-		if ( 'publish' === $new_status && 'publish' !== $old_status ) {
-			$this->just_published[ $post->ID ] = true;
-		}
 
-		$this->previous_status[ $post->ID ] = $old_status;
+
+		if ( ! isset( $this->sync_items[ $post->ID ] ) ) {
```

Comment:
I think we can make the code even more accessible if we make an alias method that describes what we're actually testing for here. That is, instead of `if ( ! isset( $this->sync_items[ $post->ID ] ) ) {`, something like `if ( $this->is_new_post( $post->ID ) ) {`